vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -330,6 +333,47 @@ func TestAccVSphereVirtualMachine_noPanicShutdown(t *testing.T) {
 	})
 }
 
+func TestAccVSphereVirtualMachine_alwaysPoweredOn(t *testing.T) {
+	var vm virtualMachine
+	var state *terraform.State
+	basic_vars := setupTemplateBasicBodyVars()
+	config := basic_vars.testSprintfTemplateBody(testAccCheckVSphereVirtualMachineConfig_really_basic)
+
+	log.Printf("[DEBUG] template= %s", testAccCheckVSphereVirtualMachineConfig_really_basic)
+	log.Printf("[DEBUG] template config= %s", config)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testBasicPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVSphereVirtualMachineDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					copyStatePtr(&state),
```

Comment:
If we didn't do a pointer to a pointer with the above comment we could just pass `state` instead of `&state`. Unless I'm missing something

Reply:
Yeah, the double pointer is necessary because a `terrafrom.State` instance is always passed by reference. So we need to directly modify the value of the pointer, or else it's essentially passed by value and does not get retained. [Example of a failure of a pass by value function](https://play.golang.org/p/IevcXmWnxY). Unless I'm missing something. :)

A different approach would be to use `State.Copy`, but I don't necessarily want a _copy_ here, but instead the in-flight state.